x-pack/test/security_api_integration/plugins/features_provider/server/index.ts

```diff
@@ -0,0 +1,384 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { PluginInitializer, Plugin, CoreSetup } from '@kbn/core/server';
+import { DEFAULT_APP_CATEGORIES } from '@kbn/core/server';
+import { FeaturesPluginSetup, FeaturesPluginStart } from '@kbn/features-plugin/server';
+
+export interface PluginSetupDependencies {
+  features: FeaturesPluginSetup;
+}
+
+export interface PluginStartDependencies {
+  features: FeaturesPluginStart;
+}
+
+export const plugin: PluginInitializer<void, void> = async (): Promise<
+  Plugin<void, void, PluginSetupDependencies, PluginStartDependencies>
+> => ({
+  setup: (_: CoreSetup<PluginStartDependencies>, deps: PluginSetupDependencies) => {
+    // Case #1: feature A needs to be renamed to feature B. It's unfortunate, but the existing feature A
+    // should be deprecated and re-created as a new feature with the same privileges.
+    case1FeatureRename(deps);
```

Comment:
Might we want to set some standard on this renaming? Something as simple as `featureAlpha` => `featureAlphaV2` ?? `featureAlpha_v2` ?? `featureAlpha_8-16` (release version) ...just so we can have general consistency?

Reply:
I think the change you’re working on right now might serve as a precedent. I do like the `_v2` suffix (or just `v2` like in the existing `fleetv2`), as it gives a sense of how many iterations the feature has gone through, but I’m open to other suggestions. In general, the suffix is needed when the successor feature is supposed to function almost exactly like the deprecated one, doesn't change its name and looks almost like the the deprecated one from the user perspective. However, when changes in behavior are significant — such as when the feature is split into two separate ones — a brand new ID that more accurately describes the new feature is preferable.